mlir/lib/Conversion/GPUToLLVMSPV/GPUToLLVMSPV.cpp

```diff
@@ -341,13 +346,21 @@ struct GPUSubgroupOpConversion final : ConvertOpToLLVMPattern<SubgroupOp> {
 
     Operation *moduleOp =
         op->template getParentWithTrait<OpTrait::SymbolTable>();
-    Type resultType = rewriter.getI32Type();
+    Type resultTy = rewriter.getI32Type();
     LLVM::LLVMFuncOp func =
-        lookupOrCreateSPIRVFn(moduleOp, funcName, {}, resultType,
+        lookupOrCreateSPIRVFn(moduleOp, funcName, {}, resultTy,
                               /*isMemNone=*/false, /*isConvergent=*/false);
 
     Location loc = op->getLoc();
     Value result = createSPIRVBuiltinCall(loc, rewriter, func, {}).getResult();
+
```

Comment:
You can query index type in the body of the function via:

```cpp
Type indexTy = getTypeConverter()->getIndexType();
```

No need for complex infrastructure for that

Reply:
Also, can we not assert on unexpected index bitwidth?

Reply:
> Also, can we not assert on unexpected index bitwidth?

Can you clarify this please

Reply:
I mean if we wanna allow other bitwidths (or simply fail instead of asserting). Although OCL-SPV only allows 32 and 64 as of now, it'd be more future-proof.

Reply:
Oh, I think I get you. Failure is better than an assert.